src/Facets/AllBridgeFacet.sol

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.7;
+
+import { ILiFi } from "../Interfaces/ILiFi.sol";
+import { LibAsset, IERC20 } from "../Libraries/LibAsset.sol";
+import { LibDiamond } from "../Libraries/LibDiamond.sol";
+import { SwapperV2 } from "../Helpers/SwapperV2.sol";
+import { ReentrancyGuard } from "../Helpers/ReentrancyGuard.sol";
+import { Validatable } from "../Helpers/Validatable.sol";
+import { LibSwap } from "../Libraries/LibSwap.sol";
+
+/// @title AllBridge Messenger Protocol Enum
+enum MessengerProtocol {
+    None,
+    Allbridge,
+    Wormhole,
+    LayerZero
+}
+
+/// @title AllBridge Interface
+interface IAllBridge {
+    function pools(bytes32 addr) external returns (address);
+
+    function swapAndBridge(
+        bytes32 token,
+        uint256 amount,
+        bytes32 recipient,
+        uint8 destinationChainId,
+        bytes32 receiveToken,
+        uint256 nonce,
+        MessengerProtocol messenger
+    ) external payable;
+}
+
+/// @title Allbridge Facet
+/// @author Li.Finance (https://li.finance)
+/// @notice Provides functionality for bridging through AllBridge
+contract AllBridgeFacet is ILiFi, ReentrancyGuard, SwapperV2, Validatable {
+    /// @notice The contract address of the AllBridge router on the source chain.
+    IAllBridge public immutable allBridge;
+
+    /// @notice The struct for the AllBridge data.
+    /// @param fees The amount of token to pay the messenger and the bridge
+    /// @param recipient The address of the token receiver after bridging.
+    /// @param destinationChainId The destination chain id.
+    /// @param receiveToken The token to receive on the destination chain.
+    /// @param nonce A random nonce to associate with the tx.
+    /// @param messenger The messenger protocol enum
+    struct AllBridgeData {
+        uint256 fees;
+        bytes32 recipient;
+        uint8 destinationChainId;
+        bytes32 receiveToken;
+        uint256 nonce;
+        MessengerProtocol messenger;
+    }
+
+    /// @notice Initializes the AllBridge contract
+    /// @param _allBridge The address of the AllBridge contract
+    constructor(IAllBridge _allBridge) {
+        allBridge = _allBridge;
+    }
+
+    /// @notice Bridge tokens to another chain via AllBridge
+    /// @param _bridgeData The bridge data struct
+    function startBridgeTokensViaAllBridge(
+        ILiFi.BridgeData memory _bridgeData,
```

Comment:
_bridgeData can be "calldata" instead of memory to save gas

Reply:
I remember vaguely that the reason why `BridgeData` is memory is because it mutated in `depositAsset`. This is similar to AcrossFacet which also requires `BridgeData` to be in memory.